stdlib/public/Distributed/LocalTestingDistributedActorSystem.swift

```diff
@@ -0,0 +1,278 @@
+  }
+
+  public func remoteCallVoid<Act, Err>(
+    on actor: Act,
+    target: RemoteCallTarget,
+    invocation: inout InvocationEncoder,
+    throwing errorType: Err.Type
+  ) async throws
+    where Act: DistributedActor,
+          Act.ID == ActorID,
+          Err: Error {
+    fatalError("Attempted to make remote call on actor \(actor) in a local-only actor system")
+  }
+
+  private struct ActorIDProvider {
+    private var counter: Int = 0
+    private let counterLock = _Lock()
+
+    init() {}
+
+    mutating func next() -> LocalTestingActorAddress {
+      let id: Int = self.counterLock.withLock {
+        self.counter += 1
+        return self.counter
+      }
+      return LocalTestingActorAddress(parse: "\(id)")
+    }
+  }
+}
+
+public struct LocalTestingInvocationEncoder: DistributedTargetInvocationEncoder {
+  public typealias SerializationRequirement = Codable
+
+  public mutating func recordGenericSubstitution<T>(_ type: T.Type) throws {
+    fatalError("Attempted to call encoder method in a local-only actor system")
+  }
+
+  public mutating func recordArgument<Argument: SerializationRequirement>(_ argument: Argument) throws {
+    fatalError("Attempted to call encoder method in a local-only actor system")
+  }
+
+  public mutating func recordErrorType<E: Error>(_ type: E.Type) throws {
+    fatalError("Attempted to call encoder method in a local-only actor system")
+  }
+
+  public mutating func recordReturnType<R: SerializationRequirement>(_ type: R.Type) throws {
+    fatalError("Attempted to call encoder method in a local-only actor system")
+  }
+
+  public mutating func doneRecording() throws {
+    fatalError("Attempted to call encoder method in a local-only actor system")
+  }
+}
+
+public class LocalTestingInvocationDecoder : DistributedTargetInvocationDecoder {
+  public typealias SerializationRequirement = Codable
+
+  public func decodeGenericSubstitutions() throws -> [Any.Type] {
+    fatalError("Attempted to call decoder method in a local-only actor system")
+  }
+
+  public func decodeNextArgument<Argument: SerializationRequirement>() throws -> Argument {
+    fatalError("Attempted to call decoder method in a local-only actor system")
+  }
+
+  public func decodeErrorType() throws -> Any.Type? {
+    fatalError("Attempted to call decoder method in a local-only actor system")
+  }
+
+  public func decodeReturnType() throws -> Any.Type? {
+    fatalError("Attempted to call decoder method in a local-only actor system")
+  }
+}
+
+// === errors ----------------------------------------------------------------
+
+@available(SwiftStdlib 5.7, *)
+public struct LocalTestingDistributedActorSystemError: DistributedActorSystemError {
+  public let message: String
+
+  public init(message: String) {
+    self.message = message
+  }
+}
+
+// === lock ----------------------------------------------------------------
+
+fileprivate class _Lock {
```

Comment:
```suggestion
@available(SwiftStdlib 5.7, *)
fileprivate class _Lock {
```

should help fix the mac CI